conjure-undertow-annotations/src/main/java/com/palantir/conjure/java/undertow/annotations/ParamDecoders.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.undertow.annotations;
+
+import com.google.common.collect.Iterables;
+import com.palantir.logsafe.exceptions.SafeIllegalArgumentException;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public final class ParamDecoders {
```

Comment:
Instead of doing the abstractions in ParamDecoders, it might be simpler to spend the 30min and add an explicit implementation for all the decoder-types we care about. It's more verbose but probably easier to follow along when looking at the generated code?

Reply:
Agreed, that's how we do it in the dialogue implementation: https://github.com/palantir/dialogue/blob/develop/dialogue-annotations/src/main/java/com/palantir/dialogue/annotations/DefaultParameterSerializer.java